Sources/SotoCore/HTTP/AsyncHTTPClient.swift

```diff
@@ -27,103 +27,41 @@ extension AsyncHTTPClient.HTTPClient {
+        httpRequest.method = request.method
+        httpRequest.headers = request.headers
+        httpRequest.body = requestBody
 
-    func execute(
-        request: AWSHTTPRequest,
-        timeout: TimeAmount,
-        on eventLoop: EventLoop,
-        logger: Logger,
-        stream: @escaping AWSResponseStream
-    ) -> EventLoopFuture<AWSHTTPResponse> {
-        let requestBody: AsyncHTTPClient.HTTPClient.Body?
-        if case .byteBuffer(let body) = request.body.payload {
-            requestBody = .byteBuffer(body)
-        } else {
-            requestBody = nil
-        }
-        do {
-            let asyncRequest = try AsyncHTTPClient.HTTPClient.Request(
-                url: request.url,
-                method: request.method,
-                headers: request.headers,
-                body: requestBody
-            )
-            let delegate = AWSHTTPClientResponseDelegate(host: asyncRequest.host, stream: stream)
-            return self.execute(
-                request: asyncRequest,
-                delegate: delegate,
-                eventLoop: .delegate(on: eventLoop),
-                deadline: .now() + timeout,
-                logger: logger
-            ).futureResult
-        } catch {
-            return eventLoopGroup.next().makeFailedFuture(error)
-        }
+        let response = try await self.execute(httpRequest, timeout: timeout, logger: logger)
+        return .init(
+            status: response.status,
+            headers: response.headers,
+            body: .init(response.body, length: nil)
+        )
     }
-}
 
-extension AsyncHTTPClient.HTTPClient.Response: AWSHTTPResponse {}
-
-@available(macOS 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
-extension AsyncHTTPClient.HTTPClient {
-    /// Execute HTTP request
-    /// - Parameters:
-    ///   - request: HTTP request
-    ///   - timeout: If execution is idle for longer than timeout then throw error
-    ///   - eventLoop: eventLoop to run request on
-    /// - Returns: EventLoopFuture that will be fulfilled with request response
     func execute(
         request: AWSHTTPRequest,
         timeout: TimeAmount,
-        on eventLoop: EventLoop? = nil,
-        logger: Logger
-    ) async throws -> AWSHTTPResponse {
-        let eventLoop = eventLoop ?? self.eventLoopGroup.any()
-        return try await self.execute(request: request, timeout: timeout, on: eventLoop, logger: logger).get()
-    }
-
-    func execute(
-        request: AWSHTTPRequest,
-        timeout: TimeAmount,
-        on eventLoop: EventLoop? = nil,
+        on eventLoop: EventLoop,
         logger: Logger,
         stream: @escaping AWSResponseStream
-    ) async throws -> AWSHTTPResponse {
-        let eventLoop = eventLoop ?? self.eventLoopGroup.any()
-        return try await self.execute(request: request, timeout: timeout, on: eventLoop, logger: logger, stream: stream).get()
+    ) async throws -> HTTPClientResponse {
+        preconditionFailure("Not supported")
```

Comment:
Should we just remove this if it's not supported?

Reply:
It will get removed in the next PR. Currently it a stub used by another section of code